### DISCUSSION TITLE
Executable not working in older GLIBC machines

### DISCUSSION THREAD

**Discussion by the-nice-guy-0** [libbpf/libbpf-bootstrap#201] [Q&A]
Thanks for this amazing tool.
I compiled it in my Ubuntu with GLIBC_2.35 and it is working.
When I try to run the same executable in an old machine with GLIBC_2.17, it is showing 
**"/lib64/libc.so.6: version `GLIBC_2.34' not found"**

Isn't libbpf CO-RE, and supposed to run it in older machines?. Or is my understanding of CO-RE is wrong?
Is there any workaround for this? I won't be able install the new GLIBC on the old machine.

Thanks

**Comment by anakryiko** [comment]
I think your understanding of BPF CO-RE is wrong. This is about being able to run BPF program on multiple kernels without recompilation. GLIBC issue is user-space side and has nothing to do with BPF itself.
